### PR TITLE
Add in basic tests

### PR DESCRIPTION
**Pull request by imduffy15**
Adding in tests that bring up an embedded kafka,
create a consumer group on it and checks that
querying the consumer group results in returning
partitions successfully.

### CODE REVIEW

**Comment by marky-mark** [comment]
So I think this one is out of date and can be closed?

**Comment by imduffy15** [comment]
> So I think this one is out of date

Not out of date? Everything here is new - readme.md updated, ApiSpec.scala file, etc.

File: src/test/scala/SimpleKafkaConsumer.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+  private val thread = new Thread {
+    import scala.collection.JavaConverters._
+
+    override def run: Unit = {
+      consumer.subscribe(List(topic).asJava)
+      consumer.partitionsFor(topic)
+
+      while (running) {
```
Comment (marky-mark) [inline src/test/scala/SimpleKafkaConsumer.scala]: this looks a bit daunting...continuously checking without some kind of wait. What do you think?

**Comment by marky-mark** [comment]
ah yea your right thought you merged some of this stuff. we went off on a tangent reviewing other stuff now were back to position 1.

Left 1 comment

File: src/test/scala/SimpleKafkaConsumer.scala (hunk 1 above)
Comment (imduffy15) [inline src/test/scala/SimpleKafkaConsumer.scala]: Its the way I've seen most of the other consumers written https://github.com/search?l=Scala&q=%22org.apache.kafka.clients.consumer.KafkaConsumer%22+%22while%22&type=Code&utf8=%E2%9C%93

I derived this from the "Automatic offset committing" section of https://kafka.apache.org/0101/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html

The poll is going to block up to  2000ms by default.

Comment (marky-mark) [inline src/test/scala/SimpleKafkaConsumer.scala]: From what I can see this thread is never releasing. Where do you get the up to 2000ms?

Suppose this is ok for test environment, for production would expect some kind of time and release tho e.g. [akka kafka streams](https://github.com/akka/reactive-kafka/blob/da090f3833e46412d82b0003dc4714f728d778fd/core/src/main/scala/akka/kafka/KafkaConsumerActor.scala)

If it causes issues we'll poll and release

Comment (imduffy15) [inline src/test/scala/SimpleKafkaConsumer.scala]: The poll value that's passed into consumer.poll might be better named as timeout.

Check running
Execute poll, wait until max buffered records is reached or the specified timeout.
send records to user defined function
Check running
....

Comment (marky-mark) [inline src/test/scala/SimpleKafkaConsumer.scala]: Ah here this does make sense, I didn't read it properly

**Pull request merged**